account_bank_statement_import_camt/models/account_bank_statement_import.py

```diff
@@ -28,15 +13,49 @@ class AccountBankStatementImport(models.TransientModel):
     """Add process_camt method to account.bank.statement.import."""
     _inherit = 'account.bank.statement.import'
 
-    def _parse_file(self, cr, uid, data_file, context=None):
+    @api.model
+    def _parse_file(self, data_file):
         """Parse a CAMT053 XML file."""
-        parser = self.pool['account.bank.statement.import.camt.parser']
+        parser = self.env['account.bank.statement.import.camt.parser']
         try:
             _logger.debug("Try parsing with camt.")
             return parser.parse(data_file)
         except ValueError:
             # Not a camt file, returning super will call next candidate:
-            _logger.debug("Statement file was not a camt file.",
-                          exc_info=True)
+            _logger.debug("Statement file was not a camt file.", exc_info=True)
             return super(AccountBankStatementImport, self)._parse_file(
-                cr, uid, data_file, context=context)
+                data_file)
+
+    @api.model
+    def _complete_statement(self, stmt_vals, journal_id, account_number):
```

Comment:
this looks complicated. Is this really something we need to decide per journal? Otherwise, the way #174 does it seems much more straightforward to me

Reply:
This was discussed with the customer, but I forgot the exact reason. Have asked him to remind us.

Reply:
@thomaspaulb any news from your customer?

Reply:
Yes: for one bank (ING), a batch payment is given and then they get back a CAMT with 1 line per batch. For another bank (Triodos), the batch comes back with all transactions separately, which is always around 100 payment which then have to be reconciled all 100 with the original batch. That is a lot of extra work. So for that bank, they use the checkbox.